### DISCUSSION TITLE
Image placement edge cases

### DISCUSSION THREAD

**Discussion by sergei-grechanik** [kovidgoyal/kitty#6979] [General]
There are a couple of edge cases when creating a placement, and the behavior doesn't seem to be documented, so I just want to confirm whether the current behavior is actually correct.

1) When both `r` and `c` are specified, the image is stretched to occupy a `c`x`r` rectangle of cells. The edge case is when only `r` or `c` is specified: in this case the missing dimension is computed based on the corresponding image dimension in pixels, similarly to how it's done when neither `r` nor `c` is specified. The rest is the same.
![image](https://github.com/kovidgoyal/kitty/assets/1084979/ec2e7655-e978-4ac2-9ce1-172c186b99a0)

2) When specifying the source rectangle with `x,y,w,h`, if the bottom-right corner is out of bounds, it is truncated to the bottom-right corner of the image (personally, I expected it to create a large additional empty space).
![image](https://github.com/kovidgoyal/kitty/assets/1084979/90a2fdca-4169-493c-b41a-1e082f0588d0)

The script to reproduce:
```shell
set -v
encoded_filename=$(printf '%s' "$(realpath ./logo/kitty.png)" | base64 -w0)
printf '\e_Gq=1,a=t,i=42,f=100,t=f;%s\e\\' "$encoded_filename"

# No row/column attributes specified
printf '\e_Gq=1,a=p,i=42\e\\\n'
# Both c and r are specified, the image will be stretched
printf '\e_Gq=1,a=p,i=42,r=2,c=50\e\\\n'
# Only c is specified. The number of rows will be computed as if c/r were not specified at all
printf '\e_Gq=1,a=p,i=42,c=50\e\\\n'
# Only r is specified, similar.
printf '\e_Gq=1,a=p,i=42,r=2\e\\\n'

# x,y are specified, no w,h
printf '\e_Gq=1,a=p,i=42,x=100,y=100\e\\end\n'
# x,y are specified, w,h are oob
printf '\e_Gq=1,a=p,i=42,x=100,y=100,w=400,h=400\e\\end\n'
# end
```

**Comment by kovidgoyal** [comment]
Yes, the current behavior is expected. I have added some text to the spec for it.
